mlflow/tracking/__init__.py

```diff
@@ -411,3 +411,39 @@ def _get_git_commit(path):
         return commit
     except (InvalidGitRepositoryError, GitCommandNotFound, ValueError, NoSuchPathError):
         return None
+
+
+def _get_git_url(uri):
+    """
+    Return the git repo path github_url#sub_directory from the uri input
+    :param uri: the expanded uri
+    :return: the git repo path included sub directory if the uri is part of git repo,
+             otherwise return the input as output
+    """
```

Comment:
Is there any way we can add a test to this function? Take a look at `tests/projects/test_projects.py` to see some tests that are using Git already (and creating small local Git repos for that purpose). You might be able to check this in one of those or just copy it and add a new test.

Reply:
Added a new test case for local git repo